### PR TITLE
executor: support load data with ignore lines

### PR DESCRIPTION
**Pull request by mccxj**
### What problem does this PR solve? 

For https://github.com/pingcap/tidb/issues/7476

### What is changed and how it works?

- support ‘IGNORE NUM LINES’ in parser.y
- check IgnoreLines in LoadDataInfo#InsertData

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported variable/fields change

### CODE REVIEW

**Comment by shenli** [comment]
@mccxj Thanks!

File: parser/parser.y [hunk 1]
```diff
@@ -809,6 +809,7 @@ import (
 	TableOptimizerHintOpt	"Table level optimizer hint"
 	TableOptimizerHints	"Table level optimizer hints"
 	TableOptimizerHintList	"Table level optimizer hint list"
+	IgnoreLines     "Ignore num(int) lines"
```
Comment (shenli) [inline parser/parser.y]: Keep alignment.

File: executor/write_test.go [hunk 2]
```diff
@@ -1635,6 +1635,26 @@ func (s *testSuite) TestLoadDataSpecifiedColumns(c *C) {
...
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test; drop table if exists load_data_test;")
+	tk.MustExec("CREATE TABLE load_data_test (id INT NOT NULL PRIMARY KEY, value TEXT NOT NULL) CHARACTER SET utf8")
+	tk.MustExec("load data local infile '/tmp/nonexistence.csv' into table load_data_test ignore 1 lines")
+	ctx := tk.Se.(sessionctx.Context)
+	ld, ok := ctx.Value(executor.LoadDataVarKey).(*executor.LoadDataInfo)
+	c.Assert(ok, IsTrue)
+	defer ctx.SetValue(executor.LoadDataVarKey, nil)
+	c.Assert(ld, NotNil)
+	// test escape
```
Comment (shenli) [inline executor/write_test.go]: What do you mean by "test escape"?

Comment (mccxj) [reply executor/write_test.go]: copy comment by mistake.

File: parser/parser.y [hunk 3]
```diff
@@ -802,6 +802,7 @@ import (
 	OptBinMod		"Optional BINARY mode"
 	OptCharset		"Optional Character setting"
 	OptCollate		"Optional Collate setting"
+	IgnoreLines     "Ignore num(int) lines"
```
Comment (jackysp) [inline parser/parser.y]: Please keep alignment.

Comment (mccxj) [reply parser/parser.y]: fixed.

File: parser/parser.y (hunk 1 above)
Comment (mccxj) [reply parser/parser.y]: fixed.

File: parser/parser.y [hunk 4]
```diff
@@ -6902,6 +6904,15 @@ LoadDataStmt:
 		$$ = x
 	}
 
+IgnoreLines:
+    {
+        $$ = uint64(0)
+    }
+|   "IGNORE" NUM "LINES"
```
Comment (shenli) [inline parser/parser.y]: In MySQL, a negative number will casue syntax error. Could you add a test case about this?

> mysql> load data local infile '/tmp/t.csv' into table t ignore -1 lines;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '-1 lines' at line 1
mysql> load data local infile '/tmp/t.csv' into table t ignore 1 lines;
ERROR 1148 (42000): The used command is not allowed with this MySQL version
mysql> quit

Comment (shenli) [reply parser/parser.y]: Got it.

**Comment by shenli** [comment]
@mccxj Thanks!

**Pull request merged**